Fix upgrade and build

arnaud-morvan opened the pull request.

sbrunner commented on geoportal/c2cgeoportal_geoportal/scaffolds/create/env.project:
```diff
@@ -0,0 +1,3 @@
+# Custom project values
+TILEGENERATION_SQS_QUEUE=<queue_name>
+TILEGENERATION_S3_BUCKET=<bucket_name>
```
Shouldn't we put the Postgres configuration in this file?

sbrunner commented on scripts/upgrade:
```diff
@@ -40,15 +40,15 @@ parser.add_argument("step", nargs="?", help="the step to run")
 parser_finalise.add_argument("--finalise", action="store_true", help="finalise the upgrade")
 parser_finalise.add_argument("--help", action="help", help="show this help message and exit")
-parser_finalise.add_argument("build_arg", nargs="*", help="a build argument")
+parser_finalise.add_argument("build_args", nargs="*", help="build arguments ex: -- --pgschema main env.default env.project")
```
`build_arg` is better for the --help ...
=>
[build_arg [build_arg ...]]

sbrunner commented on geoportal/c2cgeoportal_geoportal/scaffolds/create/build_tmpl:
```diff
@@ -22,9 +22,9 @@ parser = argparse.ArgumentParser(description="Build the project")
 parser.add_argument("--config", action="store_true", help="Build only the configuration image")
 parser.add_argument("--geoportal", action="store_true", help="Build only the geoportal image")
-parser.add_argument("--pgschema", default="main", help="The main schema name used in PostgreSQL")
+parser.add_argument("--pgschema", required=True, help="The main schema name used in PostgreSQL")
```
Shouldn't we get the PGSCHEMA from the env files?

arnaud-morvan commented:
@sbrunner : I've refactored a little the build script (added a main method).
Now schema is taken from environement and if env_files are not provided, it reuse the previous ones.
I also added some lines in local.mk to be able to test my work.

sbrunner commented:
You can ignore the Pull request title :-)

sbrunner commented on geoportal/c2cgeoportal_geoportal/scaffolds/create/build_tmpl:
```diff
@@ -18,86 +18,99 @@ def call(verbose, arguments, **kwargs):
-    sys.exit(0)
+        sys.exit(0)
+
+    env_files = args.env_files or load_env(['.env'])['ENV_FILES'].split(" ")
```
Why the or ...?

arnaud-morvan replied:
This is a default value
if args.env_files != [] => args.env_files
else load_env(['.env'])['ENV_FILES'].split(" ")

arnaud-morvan replied:
if user do not give any env_files, use previous ones.

sbrunner commented:
@arnaud-morvan fixed with a new commit, For me we can merge it as it :-)

arnaud-morvan commented:
@sbrunner : merged, thanks for the fix
